From 173fdaa92e59eff35eb2897228be685cc21e57bd Mon Sep 17 00:00:00 2001
From: Behan Webster <behanw@converseincode.com>
Date: Fri, 28 Sep 2012 15:24:42 -0400
Subject: [PATCH 4/5] Remove VLAIS from the libcrc32c code

The use of variable length arrays in structures (VLAIS) in the Linux Kernel
code precludes the use of compilers which don't implement VLAIS (for instance
the Clang compiler). Since ctx is always 32-bit, hard coding a size of 4
accomplishes the same thing without the use of VLAIS.

Signed-off-by: Mark Charlebois <charlebm@gmail.com>
Reviewed-by: Behan Webster <behanw@converseincode.com>
---
 lib/libcrc32c.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/libcrc32c.c b/lib/libcrc32c.c
index 244f548..5c47a93 100644
--- a/lib/libcrc32c.c
+++ b/lib/libcrc32c.c
@@ -43,7 +43,7 @@ u32 crc32c(u32 crc, const void *address, unsigned int length)
 {
 	struct {
 		struct shash_desc shash;
-		char ctx[crypto_shash_descsize(tfm)];
+		char ctx[4];
 	} desc;
 	int err;
 
-- 
1.7.9.5

